source/agora/consensus/EnrollmentManager.d

```diff
@@ -568,6 +569,120 @@ public class EnrollmentManager
         return true;
     }
 
+    /***************************************************************************
+
+        Check if a pre-image exists
+
+        Params:
+            enroll_key = The key for the enrollment in which the pre-image is
+                contained.
+            height = The block height of the preimage to check existence
+
+        Returns:
+            true if the pre-image exists
+
+    ***************************************************************************/
+
+    public bool hasPreimage (const ref Hash enroll_key, ulong height) @trusted
```

Comment:
Ok, so when calling `addPreimage` the function is `nothrow` and can fail so it returns a status code. But if `hasPreimage` fails it will throw an exception. This is an inconsistent API. `hasPreimage` should be `nothrow` too.